### ISSUE TITLE
Genes skipped because first letter in gene name matches letter in 

### ISSUE THREAD

**Issue by dannyn66** [CGATOxford/UMI-tools#256]
https://github.com/CGATOxford/UMI-tools/blob/b3d04590c9aa893ac9a146f16e1dec934c31ac80/umi_tools/umi_methods.py#L1432

Line 1432 of umi_methods.py is attempting to match the gene_tag of each read to the strings '__' or 'Unassigned' using the regular expression '^[__|Unassigned]' stored in self.options.skip_regex. Because of the brackets, rather than parenthesis, this statement ends up evaluating as True for any gene_tag that begins with any of the letters in the set ('U','n','a','s','i','g','e','d'). This leads to these reads being tossed. In my case, for example, any read that aligned to the gene 'elav' gets tossed. I believe you want to match the string 'Unassigned' exactly by replacing the brackets with paretheses.

**Comment by TomSmithCGAT** [comment]
Hmm, this should have been resolve in #231. What version are you using?

**Comment by TomSmithCGAT** [comment]
Stupid me. It looks like the latest version on `conda` and `pip` is 4 months old. I'll try and get around to updating this later in the week. In the meantime, you could install directly from the github repo. if you're in a rush?

**Comment by dannyn66** [comment]
I indeed used pip to install. Didn't think to use the github repo version directly. Thanks!

**Comment by TomSmithCGAT** [comment]
Yeah this is my fault for not keeping the pip version up to date. I'll leave this open as a (2nd!) reminder to do so...
